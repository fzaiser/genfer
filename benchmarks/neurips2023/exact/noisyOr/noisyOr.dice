let n0  = flip 0.5 in
let n4  = flip 0.5 in
let n1  = if n0 then flip 0.8 else flip 0.1 in
let n21  = if n0 then flip 0.8 else flip 0.1 in
let n22  = if n4 then flip 0.8 else flip 0.1 in
let n33  = if n4 then flip 0.8 else flip 0.1 in
let n2  = (n21 || n22) in
let n31  = if n1 then flip 0.8 else flip 0.1 in
let n32  = if n2 then flip 0.8 else flip 0.1 in
let n3  = (n31  || (n32 || n33)) in
n3
