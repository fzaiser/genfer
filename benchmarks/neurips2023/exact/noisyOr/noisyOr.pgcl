nat n0;
nat n1;
nat n2;
nat n3;
nat n4;
nat n21;
nat n22;
nat n33;
nat n31;
nat n32;


n0 := bernoulli(1/2);
n4 := bernoulli(1/2);
if (n0 = 1) {
  n1 := bernoulli(4/5);
  n21 := bernoulli(4/5);
} else {
  n1 := bernoulli(1/10);
  n21 := bernoulli(1/10);
}
if (n4 = 1) {
  n22 := bernoulli(4/5);
  n33 := bernoulli(4/5);
} else {
  n22 := bernoulli(1/10);
  n33 := bernoulli(1/10);
}
if (n21 = 1 || n22 = 1) {
    n2 := 1
} else {skip}
if (n1 = 1) {
        n31 := bernoulli(4/5);
} else {
  n31 := bernoulli(1/10);
}
if (n2 = 1) {
  n32 := bernoulli(4/5);
} else {
  n32 := bernoulli(1/10);
}
if (n31 = 1 || n32 = 1 || n33 = 1) {
    n3 := 1
} else {skip}

?Pr[n3];
