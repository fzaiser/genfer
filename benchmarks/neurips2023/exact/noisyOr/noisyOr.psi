// flags: --dp
def main(){ // N20n
    n1 := 0; n21 := 0; n22 := 0; n31 := 0; n32 := 0; n33 := 0;
    n0 := flip(1/2);
    n4 := flip(1/2);
    if (n0){
        n1 = flip(4/5);
        n21 = flip(4/5);
    }else{
        n1 = flip(1/10);
        n21 = flip(1/10);
    }
    if (n4){
        n22 = flip(4/5);
        n33 = flip(4/5);
    }else{
        n22 = flip(1/10);
        n33 = flip(1/10);
    }
    n2 := n21 || n22; // TODO: make fast with ||
    if (n1){
        n31 = flip(4/5);
    }else{
        n31 = flip(1/10);
    }
    if (n2){
        n32 = flip(4/5);
    }else{
        n32 = flip(1/10);
    }
    n3 := n31 || n32 || n33; // TODO: make fast with ||
    return n3; // expected: 130307/160000·δ(1)[n3]+29693/160000·δ(0)[n3]
}
