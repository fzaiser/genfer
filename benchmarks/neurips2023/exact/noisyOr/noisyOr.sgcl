# skip integration test
n0 ~ Bernoulli(1/2);
n4 ~ Bernoulli(1/2);
if n0 = 1 {
  n1 ~ Bernoulli(4/5);
  n21 ~ Bernoulli(4/5);
} else {
  n1 ~ Bernoulli(1/10);
  n21 ~ Bernoulli(1/10);
}
if n4 = 1 {
  n22 ~ Bernoulli(4/5);
  n33 ~ Bernoulli(4/5);
} else {
  n22 ~ Bernoulli(1/10);
  n33 ~ Bernoulli(1/10);
}
if n21 = 1 or n22 = 1 {
  n2 := 1;
}
if n1 = 1 {
  n31 ~ Bernoulli(4/5);
} else {
  n31 ~ Bernoulli(1/10);
}
if n2 = 1 {
  n32 ~ Bernoulli(4/5);
} else {
  n32 ~ Bernoulli(1/10);
}
if n31 = 1 or n32 = 1 or n33 =1 {
    n3 := 1;
}

return n3;
