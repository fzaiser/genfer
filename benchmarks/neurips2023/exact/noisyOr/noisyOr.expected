0.8144187
130307/160000
